staker/calculate_pool_position_reward.gno

```diff
@@ -235,96 +227,156 @@ func CalcPoolPosition() {
+				positionLastExternal[tokenId][incentiveId] = tempLastExternalAmount
+				positionExternal[tokenId][incentiveId] = externalRewards{
+					incentiveId: incentiveId,
+					poolPath:    poolPath,
+					tokenPath:   rewardToken,
+					tokenAmount: tempLastExternalAmount + positionAmount,
 				}
 			}
-			// RCE ENDS HERE
+
+			rewardMathComputeExternalRewardAmount(tokenId, incentiveId)
+			positionsExternalLastCalculatedHeight[tokenId][incentiveId] = height
+			externalToUpdate = append(externalToUpdate, incentiveId)
 		}
+		println("[ END ] CALC__EXTERNAL, TOKEN_ID", tokenId)
+	}
+	// update all external incentives time
+	uniquExternals := removeDupeStr(externalToUpdate)
+	for _, incentiveId := range uniquExternals {
+		externalLastCalculatedTimestamp[incentiveId] = now
 	}
-	println("[ END ] CALC__EXTERNAL")
-	println()
 
 	// update flag
-	println("UPDATE FLAG")
+	// println("UPDATE FLAG")
 	lastCalculatedHeight = height
-	lastCalculatedTimestamp = time.Now().Unix()
 	lastCalculatedBalance = gnsBalance(consts.STAKER_ADDR)
-	println()
+	// println()
 
-	println("[ START ] CALC__FINAL")
+	// println("[ START ] CALC__FINAL")
 	for poolPath, _ := range poolTiers {
 		amount := poolLastTmpGns[poolPath]
-		println(" > CALC__FINAL_POOL", poolPath)
-		println(" > CALC__FINAL_amount", amount)
-		println(" > CALC__FINAL_poolCurrentBlockGns[poolPath]", poolCurrentBlockGns[poolPath])
-		println(" > CALC__FINAL_poolLastTmpGns[poolPath]", poolLastTmpGns[poolPath])
+		// println(" > CALC__FINAL_POOL", poolPath)
+		// println(" > CALC__FINAL_amount", amount)
+		// println(" > CALC__FINAL_poolCurrentBlockGns[poolPath]", poolCurrentBlockGns[poolPath])
+		// println(" > CALC__FINAL_poolLastTmpGns[poolPath]", poolLastTmpGns[poolPath])
 		if amount > 0 {
 			if poolCurrentBlockGns[poolPath] >= amount {
-				println(" > CALC__SAME_OR_LEFT")
+				// println(" > CALC__SAME_OR_LEFT")
 				poolLastTmpGns[poolPath] = poolCurrentBlockGns[poolPath] - amount
-				println(" > poolLastTmpGns[poolPath]", poolLastTmpGns[poolPath])
+				// println(" > poolLastTmpGns[poolPath]", poolLastTmpGns[poolPath])
 			} else {
-				println(" > CALC__FINAL_WRONG")
+				// println(" > CALC__FINAL_WRONG")
 				poolCurrentBlockGns[poolPath] = 0
 			}
 		} else {
-			println(" > CALC__FINAL__amount = 0")
+			// println(" > CALC__FINAL__amount = 0")
 		}
 	}
-	println("[ END ] CALC__FINAL")
-	println()
+	// println("[ END ] CALC__FINAL")
+	// println()
 
 	// clear(poolCurrentBlockGns) // gno doesn't support `clear` keyword yet
 	poolCurrentBlockGns = make(map[string]uint64)
-	println("=== [ END END END ] CALC__CalcPoolPosition")
-	println()
+	// println("=== [ END END END ] CALC__CalcPoolPosition")
+	// println()
+}
+
+func removeDupeStr(strSlice []string) []string {
+	keys := make(map[string]bool)
+	list := []string{}
+	for _, entry := range strSlice {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
+			list = append(list, entry)
+		}
+	}
+	return list
 }
```

Comment:
```suggestion
func removeDupeStr(strSlice []string) []string {
    seen := make(map[string]struct{}, len(strSlice))
    result := make([]string, 0, len(strSlice))
    
    for _, str := range strSlice {
        if _, ok := seen[str]; !ok {
            seen[str] = struct{}{}
            result = append(result, str)
        }
    }
    
    return result
}
```

Modified it to be faster and less memory allocations. Below are the benchmark results comparing the original implementation and the suggested function.

```benchmark
BenchmarkRemoveDupeStrOriginal-8   	   45552	     26543 ns/op	    2126 B/op	       8 allocs/op
BenchmarkRemoveDupeStrImproved-8   	  100945	     12028 ns/op	   57368 B/op	       3 allocs/op
PASS
```